templates/index.html

```diff
@@ -145,5 +163,53 @@ <h3>Videos</h3>
   });
 
   </script>
-
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
+  <script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/2.0.4/socket.io.js"></script>
+  <script type='module'>
+    import { Spinner } from '/static/spin/spin.js'
+    
+    /* This block is required to add a spinner if the user refreshes and the job is not done */
+    {% for job_dict in session['jobs'] %}
+      {% set job = q.fetch_job(job_dict['id']) %}
+      {% if job and job.meta.get('status') and job.meta.get('status') != 'Done' %}
+        new Spinner().spin(document.getElementById('{{ job_dict["id"] }}'));
```

Comment:
abrichr 9 days ago Member
Remember, this will result in Javascript code that looks like this:

var spinner = ...
var spinner = ...
var spinner = ...
...
It doesn't make sense to assign this to a variable if you are just going to overwrite it immediately afterwards. At the very least, please remove var spinner = . I would do it like this:

let jobs = {% jobs|safe %}
for (let job of jobs) { ... }

Reply:
I just didn't set spinner to a variable